core/state/statedb.go

```diff
@@ -111,6 +111,10 @@ type StateDB struct {
 	// Per-transaction access list
 	accessList *accessList
 
+	// Ordered storage slots to be used in predicate verification as set in the tx access list.
+	// Only set in PrepareAccessList, and un-modified through execution.
+	predicateStorageSlots map[common.Address][]common.Hash
```

Comment:
Since this is intended to be used as a byte array, can we change the type to `map[common.Address][]byte` ?

Reply:
Changed this and all uses (including the type of `PredicateFunc`) to `[]byte`. Also added a helper in `utils/bytes.go` to serialize a `[]common.Hash` to `[]byte`. Let me know if there's a better place to put that function.

Reply:
This sounds good to me, just wanted to call out that we'll need to handle stripping off leading (and or trailing?) 0 bytes when using the storage slots in the warp precompile still. The actual value will an RLP encoded list of Teleporter messages, padded to have a length that's a multiple of 32.

Reply:
That's perfect and good call out.